### ISSUE TITLE
Build fails from master.

### ISSUE THREAD

**Issue by dryajov** [aurelia/dependency-injection#126]
**I'm submitting a bug report**

* **Library Version:**
1.0.0

**Please tell us about your environment:**
* **Operating System:**
OSX 10.11.6

* **Node Version:**
6.5.0

* **NPM Version:**
3.10.7

* **JSPM**
JSPM 0.16.45

* **Browser:**
NA

* **Language:**
TypeScript 2.0

**Current behavior:**
When running `npm build` from the command line I get the error:

```
SyntaxError: /Users/m90570/code/aurelia-di/dependency-injection/dist/aurelia-dependency-injection.js: `factory` has already been exported. Exported identifiers must be unique. (406:0)
  404 | * Decorator: Specifies that the decorated item should be called as a factory function, rather than a constructor.
  405 | */
> 406 | export function factory(potentialTarget?: any): any {
      | ^
  407 |   let deco = function(target) {
  408 |     metadata.define(metadata.invoker, FactoryInvoker.instance, target);
  409 |   };
```

**Expected/desired behavior:**

* **Build should succeed.**

**Comment by EisenbergEffect** [comment]
The build actually does succeed. However this library is a point of experimentation for dual Babel and TypeScript compilation. The TS compilation is not fully functioning. That's pending some other library updates. But, the build does complete and generate the proper code. I'm closing this for now since we're tracking this as an internal maintenance and build experiment.

**Comment by dryajov** [comment]
It doesn't seem to generate the commonjs code, just the top level `aurelia-dependency-injection.d.ts` and `aurelia-dependency-injection.js` under the `dist` folder. I'm trying to work through a PR to allow `@singleton` and `@transient` to be used on methods and this is standing in the way. How do I get the build to generate the commonjs files, which as far as I can see is what package.json `main` points to?

**Comment by EisenbergEffect** [comment]
You are running `gulp build` and it isn't generating everything?

**Comment by dryajov** [comment]
Yes, I was able to generate the commonjs files by running `gulp build-ts-commonjs`, but running `gulp build` it doesn't get passed the unique identifier error that I reported above.

**Comment by EisenbergEffect** [comment]
@niieani Can you review the build you set up for this. I believe it is working correctly for me but @dryajov is having some issues. Maybe just a quick look to make sure something hasn't gotten messed up or isn't environment specific.

**Comment by dryajov** [comment]
Any news on this? @EisenbergEffect @niieani

**Comment by EisenbergEffect** [comment]
It's building fine for me so I'm not sure. I would need @niieani to look at it and see if he can see anything wrong.

**Comment by dryajov** [comment]
this was fixed with b8b017516e5861e9af4d7ab4674c3a4da76a67e9, but it was a legitimate issue as it appears.
